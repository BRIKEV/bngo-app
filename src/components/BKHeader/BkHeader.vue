<template>
  <nav class="navbar">
    <h1 class="title">{{ title }}</h1>
    <slot></slot>
  </nav>
</template>

<script>
export default {
  name: 'BkHeader',

  props: {
    title: {
      type: String,
      default: '',
    },
  },
};
</script>

<style lang="scss" scoped>
  @import "@/theme/index.scss";
  .navbar {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    min-height: calculateRem(48px);
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: $brand;
    padding: 0 20px;
    .title {
      color: $white;
      font-size: $fs-h3;
      font-family: $base-font-title;
    }
  }
</style>
