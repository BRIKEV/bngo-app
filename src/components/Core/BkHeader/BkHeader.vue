<template>
  <nav class="navbar">
    <h1
      v-if="title"
      class="title"
    >
      {{ title }}
    </h1>
    <div
      v-if="logo"
      class="logoContainer"
    >
      <img src="@/assets/BnGO_logo.svg" alt="brand">
    </div>
    <slot></slot>
  </nav>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'BkHeader',

  props: {
    title: VueTypes.string.def(''),
    logo: VueTypes.bool.def(false),
  },
};
</script>

<style lang="scss" scoped>
  @import "@/theme/index.scss";
  .navbar {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    min-height: calculateRem(48px);
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: $brand;
    padding: 0 20px;
    z-index: 100;
    .logoContainer {
      height: calculateRem(35px);
      img {
        height: 100%;
      }
    }
    .title {
      color: $white;
      font-size: $fs-h3;
      font-family: $base-font-title;
    }
  }
</style>
