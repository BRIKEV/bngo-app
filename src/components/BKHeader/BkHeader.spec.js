import { shallowMount, mount } from '@vue/test-utils';
import BkHeader from '@/components/BKHeader/BkHeader.vue';

describe('BkHeader component', () => {
  it('renders props.title when passed', () => {
    const title = 'New title';
    const wrapper = shallowMount(BkHeader, {
      propsData: { title },
    });
    expect(wrapper.text()).toMatch(title);
  });
  it('Should be render correctly', () => {
    const wrapper = mount(BkHeader);
    expect(wrapper.element).toMatchSnapshot();
  });
});
