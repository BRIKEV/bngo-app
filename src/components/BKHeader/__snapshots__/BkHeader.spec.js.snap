// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BkHeader component Should be render correctly 1`] = `
<nav
  class="navbar"
>
  <h1
    class="title"
  >
    
  </h1>
</nav>
`;
